easybuild/easyconfigs/l/librsb/librsb-1.3.0.0-GCC-10.2.0.eb

```diff
@@ -0,0 +1,31 @@
+##
+# Author:    Michele Martone <michelemartone@users.sourceforge.net>
+##
+easyblock = 'ConfigureMake'
+
+name = 'librsb'
+version = '1.3.0.0'
+
+homepage = 'https://sourceforge.net/projects/librsb/'
+description = """A shared memory parallel sparse matrix computations library for the Recursive Sparse Blocks format"""
+
+toolchain = {'name': 'GCC', 'version': '10.2.0'}
+
+source_urls = [SOURCEFORGE_SOURCE]
+sources = [SOURCE_TAR_GZ]
+checksums = ['2ac8725d1f988f57df9383ae6b0bb2ed221ec935187d31ebb62ea95ee868a790']
+
+dependencies = [
+    ('zlib', '1.2.11')
+]
+
+preconfigopts = "sed -i 's/export LIBRSB_CONFIG=.*$/export LIBRSB_CONFIG=no;export LIBRSB_CONFIG_VERSION=1.3.0;/g' configure*; sed -i 's/^	true/CPPFLAGS+=\" -I.. -DRSBPP_HAS_RSB_H=1\"/g' librsbpp/configure*;"
```

Comment:
This `sed` command is confusing:
* why act on all `configure*` files? I guess that you are only interested in `confgure`. `configure.ac` has no effect unless you manually re-run `autoreconf`.
* the match on `^	true` is prone to error, it can match multiple lines. I guess that you want to replace only line 17038 of `librsbpp/configure`. If that's the case, I suggest 
    ```
    17038s/.*/CPPFLAGS+=\" -I.. -DRSBPP_HAS_RSB_H=1\"/g
    ```

Reply:
Hi @lexming. The patch was targeted for this (1.3.0.0) release, where the configure produced by me has only one `'^<TAB>true'` line.
If you tell me there is guarantee easybuild will never run `autoreconf`, we can equally use the hardcoded line number as you suggest :-)

Reply:
(or target `configure` only instead of `configure.ac`, too)

Reply:
Easybuild will not run `autoreconf` by default. You can do so in this easyconfig by loading `Autotools` and then run `autoreconf` before configuration with `preconfigopts = "autoreconf -fi &&"`.

If you want to use the approach of either modifying the configure script, or alternatively target `configure.am` and run `autoreconf`, you should use a patch to make those modifications to the code. Those changes are a little bit too much for a `sed` command.

Regarding the issue with `rsb.h`, the end result of your `sed` command is adding the root of the build directory to `CPPFLAGS`.  You would get the same result by executing make with 
```
CPPFLAGS="${CPPFLAGS} -I/path/to/build/root" make
```
which is easier to apply in EasyBuild (and more clear than the changes to configure scripts)
```
local_cppflags =  'CPPFLAGS="${CPPFLAGS} -I%(builddir)s"'
prebuildopts = local_cppflags
preinstallopts = local_cppflags
```

So, I recommend the second solution for v1.3.0.0. If you can fix the configuration scripts, none of this will be needed in future versions.

Reply:
Indeed, the `CPPFLAGS`-based way is cleaner (and clearer); latest change on  https://github.com/easybuilders/easybuild-easyconfigs/pull/14861/commits/8d3c5c8bfc1b634e1b22724e10aabf7ad0fb521a uses this one as you suggest.